runner_test.go

```diff
@@ -1,48 +1,57 @@
 package main
 
 import (
-	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
 	"testing"
 )
 
-type fakeFlagParser struct{}
-
-func (f fakeFlagParser) StringSlice(s string) []string {
-	switch s {
-	case "tags":
-		return []string{"foo", "bar"}
-	default:
-		panic(fmt.Sprintf("fakeFlagParser does not expect argument: %s", s))
-	}
+func newTestPostServer(expectedBody string, resp string, statusCode int, t *testing.T) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, _ := ioutil.ReadAll(r.Body)
+		flagParams := string(body)
+		if flagParams != expectedBody {
+			t.Errorf("fetchRource hit wrong endpoint (wanted %v but got %v)", expectedBody, flagParams)
+		}
+		w.WriteHeader(statusCode)
+		w.Write([]byte(flagParams))
+	}))
 }
 
-func (f fakeFlagParser) Int(s string) int {
-	switch s {
-	case "smart-folder-id":
-		return 123
-	default:
-		panic(fmt.Sprintf("fakeFlagParser does not expect argument: %s", s))
-	}
+func TestRunByTags(t *testing.T) {
+	expectedBody := `{"tags":["foo","bar"]}`
+	sitesResp := "Post Request Successful"
+	ts := newTestPostServer(expectedBody, sitesResp, 200, t)
+	defer ts.Close()
+	baseURL = ts.URL
+	tempOsArgs := os.Args
+	os.Args = []string{"rainforest-cli", "-tags=foo,bar", "run"}
+	main()
+	os.Args = tempOsArgs
 }
 
-func TestMakeParams(t *testing.T) {
-	flagReturner := fakeFlagParser{}
-	params := makeParams(&flagReturner)
-
-	expectedTags := []string{"foo", "bar"}
-	actualTags := params.Tags
-
-	if expectedLen, actualLen := len(expectedTags), len(actualTags); expectedLen != actualLen {
-		t.Errorf("Wrong amount of tags. Expected %d, got %d", expectedLen, actualLen)
-	}
-
-	for i, actualTag := range actualTags {
-		if expectedTags[i] != actualTag {
-			t.Errorf("Unexpected tags parameter. Expected %s, got %s", expectedTags[1], actualTag)
-		}
-	}
+func TestRunBySmartFolder(t *testing.T) {
+	expectedBody := `{"tags":[""],"smart_folder_id":700}`
+	sitesResp := "Post Request Successful"
+	ts := newTestPostServer(expectedBody, sitesResp, 200, t)
+	defer ts.Close()
+	baseURL = ts.URL
+	tempOsArgs := os.Args
+	os.Args = []string{"rainforest-cli", "-smart_folder_id=700", "run"}
+	main()
```

Comment:
Testing `main` like this is a bit weird and not a great idea in general (except for testing super-high-level stuff like exit handling). Instead, you should move as much logic as possible out of `main` and test the sub-functions.